src/compiler/checker.ts

```diff
@@ -25756,6 +25776,37 @@ namespace ts {
             return prop !== undefined && prop.valueDeclaration && hasSyntacticModifier(prop.valueDeclaration, ModifierFlags.Static);
         }
 
+        function getSuggestedLibForNonexistentProperty(missingProperty: string, containingTypeString: string) {
+            const containingType = generalizeTypeString(containingTypeString);
+            const allNewFeatures = getScriptTargetFeatures();
+            const libTargets = Object.keys(allNewFeatures);
+            for (const libTarget of libTargets) {
+                const newFeaturesOfLib = allNewFeatures[libTarget];
+                const newFeaturesOfContainingType = newFeaturesOfLib[containingType];
+                if (newFeaturesOfContainingType) {
+                    for (const newFeature of newFeaturesOfContainingType) {
+                        const isNewFeatureTheMissingProperty = newFeature === missingProperty;
+                        if (isNewFeatureTheMissingProperty) {
+                            return libTarget;
+                        }
+                    }
+                }
+            }
+
+            function generalizeTypeString(typeString: string){
+                if (typeString.includes("[]")) {
+                    return "Array";
+                }
+                else if (typeString.includes("\"\"")) {

```

Comment:
You want to get the *apparent type* and see if you have type with the name `String`, don't look for quote marks

Reply:
I should clarify, you can even check if it's equal to the `globalStringType`

Reply:
Your first comment was the key! Working on it right now and it is passing the tests without my convoluted implementation with "includes"